js/plugins/vertexai/src/gemini.ts

```diff
@@ -94,27 +96,45 @@ export const gemini15ProPreview = modelRef({
       multiturn: true,
       media: true,
       tools: true,
+      systemRole: true,
     },
   },
   configSchema: GeminiConfigSchema,
   version: 'gemini-1.5-pro-preview-0409',
 });
 
-export const SUPPORTED_GEMINI_MODELS = {
-  'gemini-1.0-pro': geminiPro,
-  'gemini-1.5-pro-preview': gemini15ProPreview,
-  'gemini-1.0-pro-vision': geminiProVision,
+export const SUPPORTED_V1_MODELS = {
+  'gemini-pro': geminiPro,
+  'gemini-pro-vision': geminiProVision,
   // 'gemini-ultra': geminiUltra,
 };
 
-function toGeminiRole(role: MessageData['role']): string {
+export const SUPPORTED_V15_MODELS = {
+  'gemini-1.5-pro-preview': gemini15ProPreview,
+};
+
+export const SUPPORTED_GEMINI_MODELS = {
+  ...SUPPORTED_V1_MODELS,
+  ...SUPPORTED_V15_MODELS,
+};
+
+function toGeminiRole(
+  role: MessageData['role'],
+  model?: ModelReference<z.ZodTypeAny>
+): string {
   switch (role) {
     case 'user':
       return 'user';
     case 'model':
       return 'model';
     case 'system':
-      throw new Error('system role is not supported');
+      if (model && SUPPORTED_V15_MODELS[model.name]) {
+        throw new Error(
+          'system role is only supported for a single message in the first position'
+        );
```

Comment:
I don't understand the combination of this condition and this error message... is it somehow implied in the context of this function that we're not in "single message in the first position"? Doesn't look like it...

Reply:
It's not implied, you are correct. It's loosely coupled based on the fact that we pull off the first system message and leave the rest. 

Technically what could happen is we get multiple messages of role system. e.g.)

```
system: always be friendly
user: say hello
model: oh hello my very best friend in the whole world
system: but not tooooo friendly
user: say hello
...
```

Right now, we pick the "first one", and leave the rest in place. The question is, what to do w/ the rest? Would it be better just to throw right away if we detect this? It's not super clear cut.

- If we pull them out, we can't guarantee the messages will be in order (gemini expects user/model turns only).
- If we concat them all, it's not clear the prompt will behave as expected
- etc